src/vs/workbench/contrib/terminal/browser/terminal.ts

```diff
@@ -139,11 +139,21 @@ export const enum TerminalConnectionState {
 	Connected
 }
 
+export interface IDetachedXTermOptions {
+	cols: number;
+	rows: number;
+	colorProvider: IXtermColorProvider;
+	capabilities?: ITerminalCapabilityStore;
+	readonly?: boolean;
+}
+
 export interface ITerminalService extends ITerminalInstanceHost {
 	readonly _serviceBrand: undefined;
 
 	/** Gets all terminal instances, including editor and terminal view (group) instances. */
 	readonly instances: readonly ITerminalInstance[];
+	/** Gets detached terminal instances created via {@link createDetachedXterm}. */
+	readonly detachedInstances: Iterable<IXtermTerminal>;
```

Comment:
Should we standardize on "terminal instance" for ITerminalInstance and "detached xterm" for these? We could call this detachedXterms